src/installer/installer.go

```diff
@@ -243,8 +244,72 @@ func (i *installer) updateSingleNodeIgnition(singleNodeIgnitionPath string) erro
 	return nil
 }
 
-func (i *installer) writeImageToDisk(ignitionPath string) error {
-	i.UpdateHostInstallProgress(models.HostStageWritingImageToDisk, "")
+func convertToInstallerKargs(kargs []string) []string {
+	var ret []string
+	for _, karg := range kargs {
+		ret = append(ret, "--append-karg", karg)
+	}
+	return ret
+}
+
+func convertToOverwriteKargs(args []string) []string {
+	var ret []string
+	i := 0
+	for i < len(args) {
+		if args[i] == "--append-karg" && i+1 < len(args) {
+			ret = append(ret, "--karg", args[i+1])
+			i += 2
+		} else {
+			i++
+		}
+	}
+	return ret
+}
+
+func (i *installer) getEncapsulatedMC(ctx context.Context, ignitionPath string, mc **mcfgv1.MachineConfig) func() error {
+	return func() error {
+		interval := 5 * time.Second
+		var err error
+		_ = utils.WaitForPredicateWithContext(ctx, 2*time.Hour, interval, func() bool {
```

Comment:
Why do we need to spend 2 hours here?
this is an optimization if it happens to delay the installation we should drop it and carry on the installation without it.
2 hours seems way too long for fetching the ignition config.

Reply:
Actually for workers we might need to wait until bootstrapping complete.  The worker ignition needs to be retrieved from the control plane.